docs/root/version_history/current.rst

```diff
@@ -5,10 +5,8 @@ Incompatible Behavior Changes
 -----------------------------
 *Changes that are expected to cause an incompatibility if applicable; deployment changes are likely required*
 
-* grpc_stats: the default value for :ref:`stats_for_all_methods <envoy_v3_api_field_extensions.filters.http.grpc_stats.v3.FilterConfig.stats_for_all_methods>` is switched
-  from true to false, in order to avoid possible memory exhaustion due to an untrusted downstream sending a large number of unique method names. The previous
-  default value was deprecated in version 1.14.0. This only changes the behavior when the value is not set. The previous behavior can be used by setting the value
-  to true. This behavior change by be overridden by setting runtime feature `envoy.deprecated_features.grpc_stats_filter_enable_stats_for_all_methods_by_default`.
+* grpc_stats: the default value for :ref:`stats_for_all_methods <envoy_v3_api_field_extensions.filters.http.grpc_stats.v3.FilterConfig.stats_for_all_methods>` is switched from true to false, in order to avoid possible memory exhaustion due to an untrusted downstream sending a large number of unique method names. The previous default value was deprecated in version 1.14.0. This only changes the behavior when the value is not set. The previous behavior can be used by setting the value to true. This behavior change by be overridden by setting runtime feature `envoy.deprecated_features.grpc_stats_filter_enable_stats_for_all_methods_by_default`.
+* http: resolving inconsistencies between :scheme and X-Forwarded-Proto. :scheme will now be set for all HTTP/1.1 requests. Thchanges the behavior of the gRPC access logger, Wasm filters, CSRF filter and oath2 filter for HTTP/1 traffic, where :scheme was previously not set. This change also validates that for front-line Envoys (Envoys configured with `xff_num_trusted_hops` set to 0) that https schemed requests can not be sent over non-TLS connections. All behavioral changes listed here can be temporarily reverted by setting `envoy.reloadable_features.add_and_validate_scheme_header` to false.
```

Comment:
When reading just the docs, I found "https schemed requests can not be sent over non-TLS connections" a bit confusing. Perhaps rephrase to make it more clear this is on the downstream/listener side? It sounds kind of like it applies on the upstream/proxy side. The implementation makes sense to me but I'm worried users won't fully understand what is being changed.

Reply:
In thinking about this more, I'm not sure we can do the scheme blocking right now. Consider encrypted traffic being forwarded to an unencrypted backend. With the upstream scheme setting still in place, that would break. Right? I think we need to do this type of blocking at the very end of the migration and then think carefully about whether it should be a config flag or not?

Reply:
It'll only break if their upstream thinks it's an L1.
Basically as written today, this will reject https-over-http at the edge, where it should, and only affects
  http2 in the clear (who does that?)
  http absolute urls with htttps:// over http, which were previously downgraded to http and served the wrong content.
at non-edge, it'll break if your Envoy is misconfigured, to look like an edge Envoy and you're sending https traffic in the clear.

If we want to allow previously supported broken config, because hey, it worked, I can limit this to be an HTTP/1.1 check for frontline Envoys, at which point it will only affect HTTP/1.1 with absolute URLs.  I assume you'd prefer that?  I could also reinstate prior behavior where https://foo.bar/thing gets silently downgraded to http://foo.bar/thing though I think that's a bug worth fixing and I think fixing it flag guarded is the way to go. WDYT?

Reply:
> and only affects http2 in the clear (who does that?)

For mesh cases I think this happens all the time. Per below, I think I don't fully understand the desired state and what we do or don't do today. If this is only edge traffic I agree this makes sense, but I'm concerned that people may have misconfigured internal proxies that might have issues? That's my main concern, so I would like to better understand what we are doing now and want to do.